contracts/account/extensions/AccountERC7579.sol

```diff
@@ -0,0 +1,370 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {PackedUserOperation, IAccountExecute} from "@openzeppelin/contracts/interfaces/draft-IERC4337.sol";
+import {IERC1271} from "@openzeppelin/contracts/interfaces/IERC1271.sol";
+import {IERC7579Module, IERC7579Validator, IERC7579Execution, IERC7579AccountConfig, IERC7579ModuleConfig, MODULE_TYPE_VALIDATOR, MODULE_TYPE_EXECUTOR, MODULE_TYPE_FALLBACK} from "@openzeppelin/contracts/interfaces/draft-IERC7579.sol";
+import {ERC7579Utils, Mode, CallType, ExecType} from "@openzeppelin/contracts/account/utils/draft-ERC7579Utils.sol";
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import {Packing} from "@openzeppelin/contracts/utils/Packing.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+import {ERC7739Signer} from "../../utils/cryptography/ERC7739Signer.sol";
+import {AccountCore} from "../AccountCore.sol";
+
+/**
+ * @dev Extension of {AccountCore} that implements support for ERC-7579 modules.
+ *
+ * To comply with the ERC-1271 support requirement, this contract implements {ERC7739Signer} as an
+ * opinionated layer to avoid signature replayability across accounts controlled by the same key.
```

Comment:
Do we want that here. It feels to me like 7579 and 7739 are independant. Sure more devs may want to use both, but I don't hink we should enforce 7739 here.

Reply:
I missed the part of the ERC that says
> The smart account MUST implement the ERC-1271 interface

resolving